mungegithub/github/github.go

```diff
@@ -1568,7 +1569,7 @@ func (obj *MungeObject) WaitForPending(requiredContexts []string) bool {
 	timeoutChan := make(chan bool, 1)
 	done := make(chan bool, 1)
 	// Wait for the github e2e test to start
-	go timeout(prMaxWaitTime, timeoutChan)
+	go timeout(obj.config.prMaxWaitTime, timeoutChan)
```

Comment:
It isn't thread safe to read prMaxWaitTime without a lock in `WaitForPending` or `WaitForNotPending` because these functions are called by a [separate goroutine](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/submit-queue.go#L505) that the submit-queue creates. The options are updated from the main goroutine in the event of a config file update so the value could be read while it is being written.  This race can be avoided by holding the options lock to read prMaxWaitTime  in `WaitForPending` and `WaitForNotPending`. An `opts *options.Options` field will need to be added to the github client and set in RegisterOptions (see the submit-queue for an [example of this](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/submit-queue.go#L945)).